[Flaky Test] `pull-gardener-e2e-kind-ha-multi-zone` is flaky

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Gardener CI -->

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:

`pull-gardener-e2e-kind-ha-multi-zone`

**CI link**:

* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12918/pull-gardener-e2e-kind-ha-multi-zone/1965361898517958656
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12917/pull-gardener-e2e-kind-ha-multi-zone/1965361138950475776
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12921/pull-gardener-e2e-kind-ha-multi-zone/1965619116219830272
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12929/pull-gardener-e2e-kind-ha-multi-zone/1966079590816288768
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12932/pull-gardener-e2e-kind-ha-multi-zone/1966118371892137984
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12929/pull-gardener-e2e-kind-ha-multi-zone/1966515213981192192
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/13050/pull-gardener-e2e-kind-ha-multi-zone/1972932637420949504
* https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/12956/pull-gardener-e2e-kind-ha-multi-zone/1976265590939062272

**Reason for failure**:

```terminal
Ginkgo timed out waiting for all parallel procs to report back
Test suite: gardener (./test/e2e/gardener)
This occurs if a parallel process exits before it reports its results to the
Ginkgo CLI.  The CLI will now print out all the stdout/stderr output it's
collected from the running processes.  However you may not see anything useful
in these logs because the individual test processes usually intercept output to
stdout/stderr in order to capture it in the spec reports.
You may want to try rerunning your test suite with
--output-interceptor-mode=none to see additional output here and debug your
suite.
  
Output from proc 1:
  Assertion or Panic detected during tree construction
  var _ = Describe("Project Tests", Ordered, Label("Project", "default"), func() {
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/project/project_roles.go:27
    Ginkgo detected a panic while constructing the spec tree.
    You may be trying to make an assertion in the body of a container node
    (i.e. Describe, Context, or When).
    Please ensure all assertions are inside leaf nodes such as BeforeEach,
    It, etc.
    Here's the content of the panic that was caught:
    failed to create garden client set: error discovering kubernetes version: Get
    "https://api.virtual-garden.local.gardener.cloud/version": dial tcp: lookup
    api.virtual-garden.local.gardener.cloud: i/o timeout
    Learn more at: http://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
Output from proc 2:
  Assertion or Panic detected during tree construction
  var _ = Describe("Project Tests", Ordered, Label("Project", "default"), func() {
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/project/project_roles.go:27
    Ginkgo detected a panic while constructing the spec tree.
    You may be trying to make an assertion in the body of a container node
    (i.e. Describe, Context, or When).
    Please ensure all assertions are inside leaf nodes such as BeforeEach,
    It, etc.
    Here's the content of the panic that was caught:
    failed to create garden client set: error discovering kubernetes version: Get
    "https://api.virtual-garden.local.gardener.cloud/version": dial tcp: lookup
    api.virtual-garden.local.gardener.cloud: i/o timeout
    Learn more at: http://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
Output from proc 3:
  Assertion or Panic detected during tree construction
  var _ = Describe("Project Tests", Ordered, Label("Project", "default"), func() {
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/project/project_roles.go:27
    Ginkgo detected a panic while constructing the spec tree.
    You may be trying to make an assertion in the body of a container node
    (i.e. Describe, Context, or When).
    Please ensure all assertions are inside leaf nodes such as BeforeEach,
    It, etc.
    Here's the content of the panic that was caught:
    failed to create garden client set: error discovering kubernetes version: Get
    "https://api.virtual-garden.local.gardener.cloud/version": dial tcp: lookup
    api.virtual-garden.local.gardener.cloud: i/o timeout
    Learn more at: http://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
Output from proc 4:
  Assertion or Panic detected during tree construction
  var _ = Describe("Project Tests", Ordered, Label("Project", "default"), func() {
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/project/project_roles.go:27
    Ginkgo detected a panic while constructing the spec tree.
    You may be trying to make an assertion in the body of a container node
    (i.e. Describe, Context, or When).
    Please ensure all assertions are inside leaf nodes such as BeforeEach,
    It, etc.
    Here's the content of the panic that was caught:
    failed to create garden client set: error discovering kubernetes version: Get
    "https://api.virtual-garden.local.gardener.cloud/version": dial tcp: lookup
    api.virtual-garden.local.gardener.cloud: i/o timeout
    Learn more at: http://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
Output from proc 5:
  Assertion or Panic detected during tree construction
  var _ = Describe("Project Tests", Ordered, Label("Project", "default"), func() {
  /home/prow/go/src/github.com/gardener/gardener/test/e2e/gardener/project/project_roles.go:27
    Ginkgo detected a panic while constructing the spec tree.
    You may be trying to make an assertion in the body of a container node
    (i.e. Describe, Context, or When).
    Please ensure all assertions are inside leaf nodes such as BeforeEach,
    It, etc.
    Here's the content of the panic that was caught:
    failed to create garden client set: error discovering kubernetes version: Get
    "https://api.virtual-garden.local.gardener.cloud/version": dial tcp: lookup
    api.virtual-garden.local.gardener.cloud: i/o timeout
    Learn more at: http://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
** End **Could not open /logs/artifacts/test_e2e_gardener_junit.xml:
open /logs/artifacts/test_e2e_gardener_junit.xml: no such file or directory
Ginkgo ran 1 suite in 1m20.417538847s
Test Suite Failed
make[1]: *** [Makefile:449: test-e2e-local-ha-multi-zone] Error 1
make[1]: Leaving directory '/home/prow/go/src/github.com/gardener/gardener'
```

**Anything else we need to know**:

It usually manifests as this "Assertion or Panic" by Ginkgo, stating that it can't connect to the virtual garden.
Happens frequently across PRs, unrelated to the changes.

--- a/charts/gardener/provider-local/templates/coredns/networkpolicy.yaml
+++ b/charts/gardener/provider-local/templates/coredns/networkpolicy.yaml
@@ -9,9 +9,14 @@ spec:
   - from:
     # Allow traffic from the docker kind network's gateway IP. This is the source IP seen by NetworkPolicies
     # when using the nodePort port-mapping on the kind control-node node (5353:30053) used by e2e tests and if the
-    # coredns pod runs on the control-plane node.
+    # coredns pod runs on the control-plane node. This is relevant for docker desktop on mac.
     - ipBlock:
         cidr: 172.18.0.1/32
+    # Allow traffic from the docker kind network's gateway IP. This is the source IP seen by NetworkPolicies
+    # when using the nodePort port-mapping on the kind control-node node (5353:30053) used by e2e tests and if the
+    # coredns pod runs on the control-plane node. This is relevant for running on linux.
+    - ipBlock:
+        cidr: 172.18.255.1/32
     # If the coredns pod runs on another node than the control-plane node, incoming traffic on the nodePort gets SNATed
     # by kube-proxy and is then sent over the tunl0 interface to the node where the coredns pod runs. In this case, the
     # source IP seen by NetworkPolicies on the destination node is the node IP of the control-plane node (on the tunl0

